build/docker/intel-fpga-initcontainer.Dockerfile

```diff
@@ -1,73 +1,68 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
 
-ARG CLEAR_LINUX_VERSION=
+# GOLANG_BASE can be used to make the build reproducible by choosing an
+# image by its hash:
+# GOLANG_BASE=golang@sha256:9d64369fd3c633df71d7465d67d43f63bb31192193e671742fa1c26ebc3a6210
+#
+# This is used on release branches before tagging a stable version.
+# The main branch defaults to using the latest Golang base image.
+FROM golang:1.16 as builder
 
-RUN swupd update --no-boot-update ${CLEAR_LINUX_VERSION}
-RUN ldconfig
-ARG DIR=/intel-device-plugins-for-kubernetes
-ARG GO111MODULE=on
-WORKDIR $DIR
+WORKDIR /intel-device-plugins-for-kubernetes
 COPY . .
 
-RUN mkdir /install_root \
-    && swupd os-install \
-    ${CLEAR_LINUX_VERSION} \
-    --path /install_root \
-    --statedir /swupd-state \
-    --bundles=rsync \
-    --no-boot-update \
-    && rm -rf /install_root/var/lib/swupd/*
+ARG ROOT=/install_root
 
-# Build CRI Hook
-RUN cd $DIR/cmd/fpga_crihook && \
-    GO111MODULE=${GO111MODULE} go install && \
-    chmod a+x /go/bin/fpga_crihook && \
-    cd $DIR/cmd/fpga_tool && \
-    go install && \
-    chmod a+x /go/bin/fpga_tool && \
-    cd $DIR && \
-    install -D ${DIR}/LICENSE /install_root/usr/local/share/package-licenses/intel-device-plugins-for-kubernetes/LICENSE && \
-    scripts/copy-modules-licenses.sh ./cmd/fpga_crihook /install_root/usr/local/share/ && \
-    scripts/copy-modules-licenses.sh ./cmd/fpga_tool /install_root/usr/local/share/
-
-# Minimal result image
-FROM scratch as final
-COPY --from=builder /install_root /
+RUN cd cmd/fpga_crihook &&  CGO_ENABLED=0 go install && \
+    cd ../fpga_tool && CGO_ENABLED=0 go install && \
+    cd ../.. && \
+    install -D LICENSE $ROOT/usr/local/share/package-licenses/intel-device-plugins-for-kubernetes/LICENSE && \
+    scripts/copy-modules-licenses.sh ./cmd/fpga_crihook $ROOT/usr/local/share/ && \
+    scripts/copy-modules-licenses.sh ./cmd/fpga_tool $ROOT/usr/local/share/
 
 ARG SRC_DIR=/usr/local/fpga-sw
 ARG DST_DIR=/opt/intel/fpga-sw
 
-# CRI hook
+# Configure CRI hook
 ARG CRI_HOOK=intel-fpga-crihook
 ARG FPGA_TOOL=fpgatool
 ARG HOOK_CONF=$CRI_HOOK.json
 ARG HOOK_CONF_SRC=$SRC_DIR/$HOOK_CONF
 ARG HOOK_CONF_DST=$DST_DIR/$HOOK_CONF
 
-COPY --from=builder /go/bin/fpga_crihook $SRC_DIR/$CRI_HOOK
-COPY --from=builder /go/bin/fpga_tool $SRC_DIR/$FPGA_TOOL
+RUN install -D /go/bin/fpga_crihook $ROOT/$SRC_DIR/$CRI_HOOK
+RUN install -D /go/bin/fpga_tool $ROOT/$SRC_DIR/$FPGA_TOOL
 
-RUN echo -e "{\n\
+RUN echo "{\n\
     \"hook\" : \"$DST_DIR/$CRI_HOOK\",\n\
     \"stage\" : [ \"prestart\" ],\n\
     \"annotation\": [ \"fpga.intel.com/region\" ]\n\
-}\n">>$HOOK_CONF_SRC
+}\n">>$ROOT/$HOOK_CONF_SRC
 
-RUN echo -e "#!/bin/sh\n\
+RUN echo "#!/bin/sh\n\
 rsync -a --delete $SRC_DIR/ $DST_DIR\n\
 mkdir -p /etc/containers/oci/hooks.d\n\
 ln -sf $HOOK_CONF_DST /etc/containers/oci/hooks.d/$HOOK_CONF\n\
```

Comment:
we'll need to get rid of the deploy script when moving to distroless in the final image so these will need to be created under `$ROOT` too

Reply:
> That could be tricky as the script creates that symlink on the host filesystem.

Right, I forgot this. The distroless images don't have shell either so need to think how this would work...

Reply:
I can think of only two possible solutions for that:
- copy file instead of linking
- extend static utility that we're going to use to perform linking

Reply:
I'm considering another alternative that might give us more scalable option:

```
+WORKDIR /install_root/bin
+RUN curl http://landley.net/toybox/bin/toybox-x86_64 -O \
+    && chmod +x toybox-x86_64 \
+    && ln -s toybox-x86_64 sh \
+    && ln -s toybox-x86_64 ln \
+    && ln -s toybox-x86_64 cp \
+    && ln -s toybox-x86_64 rm
```

Reply:
yeah, that would work too

Reply:
can you easily give it a try?

Reply:
worked just fine. Updated the PR.